docs/concepts-advanced/environment-variables.md

```diff
@@ -1,170 +1,352 @@
-      name:"MY_OWN_REGISTRY_PASSWORD",
-      value:"MySecretPassword"}
-  ) {
-    id
-  }
-}
-```
+    ```graphql title=""
+    mutation {
+      addOrUpdateEnvVariableByName(
+        input: {
+          project: "lagoon-demo"
+          scope: GLOBAL
+          name: "JAVA_OPTS"
+          value: "-Xmx512m"
+        }
+      ) {
+        id
+        name
+        scope
+        value
+      }
+    }
+    ```
 
-### Global Environment Variables (Lagoon API)
+## Environment Files (committed to Git Repo)
 
-Global environment variables are available as both a build-time environment variable so that it may be consumed by builds, and also a runtime variable so that it is available within running containers.
+Use environment files for config that can be safely committed to a Git repository.
 
-## Environment Files (existing directly in the Git Repo)
+Environment files are only read by containers based on the [Lagoon `commons` base image](../docker-images/commons.md). They aren't read by Lagoon, but by the containers `ENTRYPOINT` scripts, which looks for them in the containers working directory. This means that these env vars are only available at runtime.
 
-If you have environment variables that can safely be saved within a Git repository, we suggest adding them directly into the Git repository in an environment file. These variables will also be available within local development environments and are therefore more portable.
+!!! info
+    If environment files aren't loading, confirm that you're using Lagoon base images, that you're copying them into the contianer as part of your `Dockerfile`, and that your container has a `WORKDIR` setting that points to the location of these files.
 
-The syntax in the environment files is as following:
+The syntax of an environment file is `ENV_VAR_NAME="env var value"`:
 
-```bash title="myenvironment.env"
-MYVARIABLENAME="MyVariableValue"
-MVARIABLENUMBER=4242
+```bash title=".env"
+PHP_MEMORY_LIMIT="400M"
+PHP_MAX_EXECUTION_TIME=900
 DB_USER=$DB_USERNAME # Redefine DB_USER with the value of DB_USERNAME e.g. if your application expects another variable name for the Lagoon-provided variables.
 ```
 
-### `.lagoon.env.$BRANCHNAME`
+### `.lagoon.env` and `.lagoon.env.BRANCHNAME`
+
+Use `.lagoon.env` when you want to always set env vars, but only for Lagoon images.
 
-If you want to define environment variables different per environment you can create a `.lagoon.env.$BRANCHNAME` e.g. for the main branch `.lagoon.env.main`. This helps you keep environment variables apart between environments.
+Use `.lagoon.env.BRANCHNAME` when you want to set different env vars per environment. `BRANCHNAME` will be replaced by the env vars `$LAGOON_GIT_BRANCH` or `$LAGOON_GIT_SAFE_BRANCH`. For example, if you deployed the branch `main`, the file name would be `.lagoon.env.main`. Branch names that contain special characters are sanitized, so check the value of `$LAGOON_GIT_SAFE_BRANCH` to determine what the filename should be.
```

Comment:
`"BRANCHNAME" will be replaced by the` 
I think that this can be stated slightly more strongly - `must be` rather than `will be` for instance

Reply:
"branchname must be replaced," sounds like a request to the reader, but this is documenting what happens in the container, so I'm not sure I would change anything, assuming I understand the feedback.